### PR TITLE
Add lua test script, minor fix to Lua API files

### PR DESCRIPTION
**Pull request by wing328**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the correct branch: `3.0.0` branch for changes related to OpenAPI spec 3.0. Default: `master`.
- [x] Copied the [technical committee](https://github.com/swagger-api/swagger-codegen/#swagger-codegen-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

- Add lua test script
- minor fix to Lua API files
- put files under a folder 

cc @daurnimator

### CODE REVIEW

File: modules/swagger-codegen/src/main/resources/lua/api.mustache [hunk 1]
```diff
@@ -8,12 +8,14 @@ local dkjson = require "dkjson"
...
 
 -- model import
-local {{{packageName}}}_{{classname}} = require "{{{packageName}}}.api.{{classname}}"
+{{#imports}}
+local {{{packageName}}}_{{import}} = require "{{{packageName}}}.model.{{import}}"
+{{/imports}}
 
-local {{{packageName}}}= {}
-local {{{packageName}}}_mt = {
+local {{{classname}}}= {}
```
Comment (daurnimator) [inline modules/swagger-codegen/src/main/resources/lua/api.mustache]: unrelated style: should add a space before `=`

Comment (daurnimator) [inline modules/swagger-codegen/src/main/resources/lua/api.mustache]: unrelated style: should add a space before `=`

File: samples/client/petstore/lua/test.lua [hunk 2]
```diff
@@ -0,0 +1,12 @@
+local petstore_client = require "petstore.api.pet_api"
```
Comment (daurnimator) [inline samples/client/petstore/lua/test.lua]: This is an example rather than what I'd call a test: intentional?

Comment (wing328) [reply samples/client/petstore/lua/test.lua]: Yup, agreed it's more like an example. I'll submit a separate PR with integration tests using https://github.com/bluebird75/luaunit

Comment (daurnimator) [reply samples/client/petstore/lua/test.lua]: Use busted, not luaunit

File: modules/swagger-codegen/src/main/resources/lua/api.mustache (hunk 1 above)
Comment (wing328) [reply modules/swagger-codegen/src/main/resources/lua/api.mustache]: Fixed via 39179a2

File: samples/client/petstore/lua/test.lua (hunk 2 above)
Comment (wing328) [reply samples/client/petstore/lua/test.lua]: Sure, I'll also add auto-generated test files for API and models.

Ref: https://github.com/Olivine-Labs/busted

**Pull request merged**